weaviate/connect/connection.py

```diff
@@ -458,6 +468,35 @@ def timeout_config(self, timeout_config: Union[Tuple[Real, Real], Real]):
     def proxies(self) -> dict:
         return self._proxies
 
+    def wait_for_weaviate(self, wait_for_weaviate: Optional[int]):
+        """
+        Waits until weaviate is ready or the timelimit given in 'wait_for_weaviate' has passed.
+
+        Parameters
+        ----------
+        wait_for_weaviate : Optional[int]
+            Describes how long the client will wait for weaviate to start in seconds.
+
+        Raises
+        ------
+        ConnectionError
+            If weaviate takes longer than the timelimit to respond.
+        """
+
+        ready_url = self.url + self._api_version_path + "/.well-known/ready"
+        for _i in range(0, wait_for_weaviate, 5):
+            try:
+                status_code = requests.get(
+                    ready_url,
+                ).status_code
+                if status_code != 200:
+                    time.sleep(5)
+                else:
+                    return
+            except requests.exceptions.ConnectionError:
+                time.sleep(5)
+        raise ConnectionError("Weaviate did not start in time.")
```

Comment:
Is this the same exception that is raised without your PR?

Reply:
Without the PR it was:
"requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))".

Reply:
Then please raise exactly the same exception, and ideally add a test for this

Simply create an integration test with an url that has no weavaite running and add a very short timeout. Then make sure that the test passes with and without "wait_for_weaviate" (please parameterize)